### PR TITLE
core/tracing: add changelog

### PR DESCRIPTION
**Pull request by s1na**

### CODE REVIEW

File: core/tracing/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+#### Removed methods
+
+The hooks `CaptureStart` and `CaptureEnd` have been removed. These hooks signaled the top-level call frame of a transaction. The relevant info will be now emitted by `OnEnter` and `OnExit` which are emitted for every call frame. They now contain a `depth` parameter which can be used to distinguish the top-level call frame when necessary. The `create bool` parameter to `CaptureStart` can now be inferred from `typ byte` in `OnEnter`, i.e. `vm.OpCode(typ) == vm.CREATE`.
+
+#### Modified methods
+
+- `CaptureTxStart` -> `OnTxStart(vm *VMContext, tx *types.Transaction, from common.Address)`. It now emits the full transaction object as well as `from` which should be used to get the sender address. The `*VMContext` is a replacement for the `*vm.EVM` object previously passed to `CaptureStart`.
+- `CaptureTxEnd` -> `OnTxEnd(receipt *types.Receipt, err error)`. It now returns the full receipt object.
+- `CaptureEnter` -> `OnEnter(depth int, typ byte, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int)`. The new `depth int` parameter indicates the call stack depth. It is 0 for the top-level call.
```
Comment (maoueh) [inline core/tracing/CHANGELOG.md:64]:
```suggestion
- `CaptureEnter` -> `OnEnter(depth int, typ byte, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int)`. The new `depth int` parameter indicates the call stack depth. It is 0 for the top-level call. Furthermore, the location where `OnEnter` is called in the EVM is now made a soon as a call is started. This means some specific error cases that were not before calling `OnEnter/OnExit` will now do so, leading some transaction to have an extra call traced.
```

**Review by holiman — APPROVED** [review]
LGTM

File: core/tracing/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Changelog
+
+All notable changes to the tracing interface will be documented in this file.
```
Comment (holiman) [inline core/tracing/CHANGELOG.md:3]: Heh, that's quite a promise you're making here :)

Comment (s1na) [reply core/tracing/CHANGELOG.md:3]: :trollface:

**Pull request merged**